packages/lib/src/storybook/StoryMeta.ts

```diff
@@ -1,20 +1,39 @@
 import { ComponentMeta } from '@storybook/react';
 import mergeWith from 'lodash/mergeWith';
 import { ComponentProps } from 'react';
+
 import { StoryArgType, storybookArgTypes } from './storybookArgTypes';
 import { storybookExcludedControlParams } from './storybookExcludedControlParams';
 
-export interface StoryMeta<T extends React.ElementType>
-  extends Omit<ComponentMeta<T>, 'component'> {
+// Re-defining LG provider prop keys here since importing from the package
+// will cause circular dependencies
+interface LeafyGreenProviderProps {
+  baseFontSize?: number;
+  darkMode?: boolean;
+}
+
+export interface StoryMeta<
+  T extends React.ElementType,
+  XP extends Record<string, any> = {},
+> extends Omit<ComponentMeta<T>, 'component' | 'argTypes' | 'args'> {
   parameters: ComponentMeta<T>['parameters'] & {
     /**
      * The default story to be displayed on `mongodb.design`
      */
     default: string;
   };
-  argTypes?: Partial<{
-    [name in keyof ComponentProps<T>]: StoryArgType;
-  }>;
+  argTypes?: Partial<
+    | {
+        [key in keyof ComponentProps<T>]: StoryArgType;
+      }
+    | {
+        [key in keyof LeafyGreenProviderProps]: StoryArgType;
+      }
+    | {
+        [key in keyof XP]: StoryArgType;
```

Comment:
Doesn't this just allow any string key now, instead of making sure the key is included in the component's props?